File: 0/Templates/CSharp/ProjectTemplates/CompositionAppTemplate/MainViewFactory.cs
using System;
using Windows.ApplicationModel.Core;

namespace $safeprojectname$
{
    public sealed class MainViewFactory : IFrameworkViewSource
    {
        IFrameworkView IFrameworkViewSource.CreateView()
        {
            return new MainView();
        }

        static int Main(string[] args)
        {
            CoreApplication.Run(new MainViewFactory());

            return 0;
        }
    }
}

File: 0/Templates/VisualBasic/ProjectTemplates/CompositionAppVBTemplate/MainViewFactory.vb
Imports Windows.ApplicationModel.Core

Public Class MainViewFactory
    Implements IFrameworkViewSource

    Public Function CreateView() As IFrameworkView Implements IFrameworkViewSource.CreateView
        Return New MainView
    End Function

    Shared Sub Main()
        CoreApplication.Run(New MainViewFactory())
    End Sub
End Class
